src/test/java/com/jcraft/jsch/JSchTest.java

```diff
@@ -1,17 +1,29 @@
 package com.jcraft.jsch;
 
+import org.junit.jupiter.api.AfterEach;
 import org.junit.jupiter.api.BeforeEach;
 import org.junit.jupiter.api.Test;
-
+import java.util.HashMap;
 import java.util.Hashtable;
-
+import java.util.Properties;
 import static org.junit.jupiter.api.Assertions.*;
 
 class JSchTest {
+    private Hashtable<String, String> orgConfig;
+    private Logger orgLogger;
 
     @BeforeEach
+    @SuppressWarnings("unchecked")
     void resetJsch() {
-        JSch.setLogger(null);
+      orgConfig = (Hashtable<String, String>) JSch.config.clone();
```

Comment:
How about simply doing `new Hashtable<>(JSch.config)`, to avoid the ugly cast and call to `clone()`?

Reply:
The "copy-constructor" creates shallow copies but with immutable values this isn't an issue here, so I've changed it as suggested.